rpc/core/blocks.go

```diff
@@ -81,6 +82,14 @@ func Block(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlock, error)
 	return &ctypes.ResultBlock{BlockID: blockMeta.BlockID, Block: block}, nil
 }
 
+// BlockByHash gets block of given hash.
+// More: https://tendermint.com/rpc/#/Info/block_by_hash
+func BlockByHash(ctx *rpctypes.Context, hash bytes.HexBytes) (*ctypes.ResultBlock, error) {
+	blockMeta := blockStore.LoadBlockMetaByHash(hash)
```

Comment:
I think it's cheaper to contruct blockmeta from loaded block than loading block meta from the DB.